src/partner-listing-block/view.ts

```diff
@@ -22,24 +22,36 @@ function getContext() {
 	return getContextRaw< DirectoryBlockContext >();
 }
 
-const { state } = store( 'a4a-partner-directory', {
+const partnerStore = store( 'a4a-partner-directory', {
 	// The server-side adds the initial `partners` state. So even though the initial state
 	// appears empty here, at runtime it will conform to this type.
 	state: {} as {
 		partners: PartnerDataExtraFields[];
 	},
+	search: function* (formData: FormData) {
+		const url = '/wp-json/a4a-partner-directory/v1/partners'
+		// @ts-ignore - I cannot instantiate URLSearchParams with FormData in TS. That's a known bug.
+		const queryParams = new URLSearchParams(formData).toString()
+		const response: Response = yield fetch( `${url}?${queryParams}` )
+		const partners: PartialPartnerData[] = yield response.json();
+
+		partnerStore.state.partners = partners.map( ( partner ) => ( {
+			partner,
+			url: getContext().detailsUrlPrefix + partner.id 
+		} ) );
+	},
 	actions: {
+		handleSearchChange: function* (event: Event) {
+			if(!event.target) {
+				return
+			}
+
+			partnerStore.search(new FormData((event.target as HTMLSelectElement).form as HTMLFormElement))
```

Comment:
I'm surprised this worked actually. When calling a generator from another generator you'd normally want to use `yield*` so that `handleSearchChange` will yield all of the elements from its "child" generator.

```suggestion
			yield* partnerStore.search(new FormData((event.target as HTMLSelectElement).form as HTMLFormElement))
```

I'd probably still do it for good measure. You'll need to make changes to `target` and `moduleResolution` in `tsconfig.json` though.

Reply:
Oh good catch. I'll change it.